rp2040-hal/src/gpio/mod.rs

```diff
@@ -128,3 +152,1262 @@ pub enum OutputOverride {
+        } else {
+            OutputSlewRate::Slow
+        }
+    }
+
+    /// Set the slew rate for the pin.
+    #[inline]
+    pub fn set_slew_rate(&mut self, rate: OutputSlewRate) {
+        self.id
+            .pad_ctrl()
+            .modify(|_, w| w.slewfast().bit(OutputSlewRate::Fast == rate));
+    }
+
+    /// Get wether the schmitt trigger (hysteresis) is enabled.
+    #[inline]
+    pub fn get_schimtt_enabled(&self) -> bool {
+        self.id.pad_ctrl().read().schmitt().bit_is_set()
+    }
+
+    /// Enable/Disable the schmitt trigger.
+    #[inline]
+    pub fn set_schmitt_enabled(&self, enable: bool) {
+        self.id.pad_ctrl().modify(|_, w| w.schmitt().bit(enable));
+    }
+
+    /// Get the state of the digital output circuitery of the pad.
+    #[inline]
+    pub fn get_output_disable(&mut self) -> bool {
+        self.id.pad_ctrl().read().od().bit_is_set()
+    }
+
+    /// Set the digital output circuitery of the pad.
+    #[inline]
+    pub fn set_output_disable(&mut self, disable: bool) {
+        self.id.pad_ctrl().modify(|_, w| w.od().bit(disable));
+    }
+
+    /// Get the state of the digital input circuitery of the pad.
+    #[inline]
+    pub fn get_input_enable(&mut self) -> bool {
+        self.id.pad_ctrl().read().ie().bit_is_set()
+    }
+
+    /// Set the digital input circuitery of the pad.
+    #[inline]
+    pub fn set_input_enable(&mut self, enable: bool) {
+        self.id.pad_ctrl().modify(|_, w| w.ie().bit(enable));
+    }
+
+    // =======================
+    // IO related methods
+
+    /// Set the input override.
+    #[inline]
+    pub fn set_input_override(&mut self, override_value: InputOverride) {
+        use pac::io_bank0::gpio::gpio_ctrl::INOVER_A;
+        let variant = match override_value {
+            InputOverride::DontInvert => INOVER_A::NORMAL,
+            InputOverride::Invert => INOVER_A::INVERT,
+            InputOverride::AlwaysLow => INOVER_A::LOW,
+            InputOverride::AlwaysHigh => INOVER_A::HIGH,
+        };
+        self.id.io_ctrl().modify(|_, w| w.inover().variant(variant));
+    }
+
+    /// Set the output enable override.
+    #[inline]
+    pub fn set_output_enable_override(&mut self, override_value: OutputEnableOverride) {
+        use pac::io_bank0::gpio::gpio_ctrl::OEOVER_A;
+        let variant = match override_value {
+            OutputEnableOverride::DontInvert => OEOVER_A::NORMAL,
+            OutputEnableOverride::Invert => OEOVER_A::INVERT,
+            OutputEnableOverride::Disable => OEOVER_A::DISABLE,
+            OutputEnableOverride::Enable => OEOVER_A::ENABLE,
+        };
+        self.id.io_ctrl().modify(|_, w| w.oeover().variant(variant));
+    }
+
+    /// Set the output override.
+    #[inline]
+    pub fn set_output_override(&mut self, override_value: OutputOverride) {
+        use pac::io_bank0::gpio::gpio_ctrl::OUTOVER_A;
+        let variant = match override_value {
+            OutputOverride::DontInvert => OUTOVER_A::NORMAL,
```

Comment:
I see it's not new, but perhaps that gpio rework is a good opportunity to change this as well: The naming `DontInvert` feels wrong. It's probably because it's a kind of double negation, and because output override can not only invert the pin. While thinking about a better name, I considered `Normal` - only later seeing that the SVD uses the same term. Is there any reason why `Normal` would not be a good choice?